eclair-core/src/main/scala/fr/acinq/eclair/api/NewService.scala

```diff
@@ -0,0 +1,391 @@
+                  } ~
+                  path("receive") {
+                    formFields("description".as[String], "amountMsat".as[Long].?, "expireIn".as[Long].?) { (desc, amountMsat, expire) =>
+                      complete(receive(desc, amountMsat, expire))
+                    }
+                  } ~
+                  path("parseinvoice") {
+                    formFields("invoice".as[PaymentRequest]) { invoice =>
+                      complete(invoice)
+                    }
+                  } ~
+                  path("findroute") {
+                    formFields("nodeId".as[PublicKey].?, "amountMsat".as[Long].?, "invoice".as[PaymentRequest].?) { (nodeId, amount, invoice) =>
+                      findRoute(nodeId, amount, invoice)
+                    }
+                  } ~
+                  path("send") {
+                    formFields("amountMsat".as[Long].?, "paymentHash".as[ByteVector32](sha256HashUnmarshaller).?, "nodeId".as[PublicKey].?, "invoice".as[PaymentRequest].?) { (amountMsat, paymentHash, nodeId, invoice) =>
+                      complete(send(nodeId, amountMsat, paymentHash, invoice))
+                    }
+                  } ~
+                  path("checkpayment") {
+                    formFields("paymentHash".as[ByteVector32](sha256HashUnmarshaller).?, "invoice".as[PaymentRequest].?) { (paymentHash, invoice) =>
+                      checkpayment(paymentHash, invoice)
+                    }
+                  } ~
+                  path("audit") {
+                    formFields("from".as[Long].?, "to".as[Long].?) { (from, to) =>
+                      complete(audit(from, to))
+                    }
+                  } ~
+                  path("networkfees") {
+                    formFields("from".as[Long].?, "to".as[Long].?) { (from, to) =>
+                      complete(networkFees(from, to))
+                    }
+                  } ~
+                  path("channelstats") {
+                    complete(channelStats())
+                  } ~
+                  path("ws") {
+                    handleWebSocketMessages(makeSocketHandler)
+                  } ~
+                  path(Segment) { _ => reject() }
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+
+  def connect(nodeId_opt: Option[PublicKey], host_opt:Option[String], port_opt: Option[Int], uri_opt: Option[String]): Route = (nodeId_opt, host_opt, port_opt, uri_opt) match {
+    case (None, None, None, Some(uri)) => complete((appKit.switchboard ? Peer.Connect(NodeURI.parse(uri))).mapTo[String])
+    case (Some(nodeId), Some(host), Some(port), None) => complete((appKit.switchboard ? Peer.Connect(NodeURI.parse(s"$nodeId@$host:$port"))).mapTo[String])
+    case _ => reject(UnknownParamsRejection("Wrong arguments for 'connect'"))
+  }
+
+  def open(nodeId: PublicKey, fundingSatoshis: Long, pushMsat: Option[Long], fundingFeerateSatByte: Option[Long], flags: Option[Int]): Future[String] = {
```

Comment:
I know this is WIP, but I think all of this stuff would be a good idea to create a trait and implement that trait in this file. 

Some work i've done before: 

https://github.com/ACINQ/eclair/pull/653/files#diff-9cda5c6eaeb9a6dfc654be6a6db44aadR15

it means it a lot easier as a reader of this code base to reason about these things. Also with this method specifically, I would _highly_ recommend implementing types for `Satoshis` and `MilliSatoshis` as passing around raw `Long` is very dangerous. 

Feel free to copy these implementations: 

https://github.com/bitcoin-s/bitcoin-s-core/blob/master/core/src/main/scala/org/bitcoins/core/currency/CurrencyUnits.scala#L64-L94

https://github.com/bitcoin-s/bitcoin-s-core/blob/master/core/src/main/scala/org/bitcoins/core/protocol/ln/currency/MilliSatoshis.scala#L16

Reply:
cACK for declaring the operations in a separate trait as it makes the code more readable and composable. I also agree on the typing the currency units but i think this should be part of a bigger PR for eclair as it needs to be refactored in many places, for example even if we had `fr.acinq.bitcoin.MilliSatoshi` in the parameters type i'd still be sending a `Long` down to the payment lifecycle actor.